dts/binding-template.yaml

```diff
@@ -207,6 +195,66 @@ properties:
         required: false
         default: [0x12, 0x34] # Same as 'uint8-array-with-default = [12 34]'
 
+# 'child-binding' can be used when a node has children that all share the same
+# properties. Each child gets the contents of 'child-binding' as its binding
+# (though an explicit 'compatible = ...' on the child node takes precedence, if
+# a binding is found for it).
+#
+# The example below is for a binding for PWM LEDs, where the child nodes are
+# required to have a 'pwms' property. It corresponds to this .dts structure
+# (assuming the binding has 'compatible: "pwm-leds"'):
+#
+# pwmleds {
+#         compatible = "pwm-leds";
+#
+#         red_pwm_led {
+#                 pwms = <&pwm3 4 15625000>;
+#         };
+#         green_pwm_led {
+#                 pwms = <&pwm3 0 15625000>;
+#         };
+#         ...
+# };
+child-binding:
+    title: PWM LED
+    description: LED that uses PWM
+
+    properties:
+        pwms:
+            type: phandle-array
+            required: true
+
+# 'child-binding' also works recursively. For example, the binding below would
+# provide a binding for the 'grandchild' node in this .dts (assuming
+# 'compatible: "foo"'):
+#
+# parent {
+#         compatible = "foo";
+#         child {
+#                 grandchild {
+#                         prop = <123>;
+#                 };
+#         };
+# }
+#
+# WARNING: Due to implementation issues with legacy code, only up to two levels
```

Comment:
Just curious, is this limitation enforced or just something about how the old code works?

Reply:
Not enforced now. The grandgrandchildren will just silently not get the binding in the old code if you try to nest it to three levels.

Could add a check for it, though I wonder if we will ever have three levels in practice.